appmap/_implementation/configuration.py

```diff
@@ -0,0 +1,147 @@
+    static = False
+    new_call_event = partial(CallEvent,
+                             defined_class=defined_class,
+                             method_id=method_id,
+                             path=path,
+                             lineno=lineno,
+                             static=static)
+
+    @wraps(func)
+    def run(*args, **kwargs):
+        if not recorder.enabled:
+            return func(*args, **kwargs)
+        call_event = new_call_event()
+        if not isinstance(call_event, Event):
+            raise TypeError
+        call_event_id = call_event.id
+        recorder.add_event(call_event)
+        try:
+            ret = func(*args, **kwargs)
+            return_event = ReturnEvent(parent_id=call_event_id)
+            if not isinstance(return_event, Event):
+                raise TypeError
+            recorder.add_event(return_event)
+            return ret
+        except:  # noqa: E722
+            recorder.add_event(ExceptionEvent(parent_id=call_event_id, exc_info=sys.exc_info))
+            raise
+    return run
+
+
+def in_set(method, which):
+    class_name, func_name = split_method_name(method)
+    logging.debug(('  class_name %s'
+                   ' funcname %s'
+                   ' which %s'),
+                  class_name,
+                  func_name,
+                  which)
+    if func_name.startswith('__'):
+        return False
+    return (class_name in which
+            or method.__qualname__ in which)
+
+class ConfigFilter(Filter):
+    includes = set()
+    excludes = set()
+
+    @classmethod
+    def load_config(cls):
+        if not enabled():
+            return cls
+
+        config_file = os.getenv("APPMAP_CONFIG", "appmap.yml")
+        with open(config_file) as file:
+            config = yaml.load(file, Loader=yaml.BaseLoader)
+            logging.debug('config %s', config)
+
+        for package in config['packages']:
+            path = package['path']
+            cls.includes.add(path)
+            if 'exclude' in package:
+                excludes = [f'{path}.{e}' for e in package['exclude']]
+                cls.excludes.update(excludes)
+        logging.debug('ConfigFilter, includes %s', cls.includes)
+        logging.debug('ConfigFilter, excludes %s', cls.excludes)
+
+        return cls
+
+    @classmethod
+    def excluded(cls, method):
+        ret = in_set(method, cls.excludes)
+        logging.debug('ConfigFilter, %s excluded? %s', method, ret)
+        return ret
+
+    @classmethod
+    def included(cls, method):
+        ret = in_set(method, cls.includes)
+        logging.debug('ConfigFilter, %s included? %s', method, ret)
+        return ret
+
+    def call(self, method):
+        if self.excluded(method):
+            return method
+        if self.included(method):
+            return wrap(method)
+        return self.next_filter.call(method)
+
+
+
+class BuiltinFilter(Filter):
+
+    @classmethod
+    def load_config(cls):
+        if not enabled:
```

Comment:
```suggestion
        if not enabled():
```

Reply:
`enabled()` has moved into `appmap._implementation.env`, and that's now what's being used here.